common/pytest-fixtures/README.md

```diff
@@ -0,0 +1,90 @@
+
+This tutorial explains how to use AML fixture with your code with this [sample](/samples). By including this python file in your code, this will set the following AML SDK mocks:
+
+* Workspace.Get
+* PythonScriptStep
+* Pipeline
+* Pipeline.publish
+* ComputeTarget (AmlCompute only)
+
+### Fixture for mocking AML SDK
+
+> [This fixture](./test_aml_mock_fixtures_default.py) is **independent** from other utils and tools used in
+[samples](/samples) of this repo.
+
+This code is for cases not using env util.
+
+1. Copy and paste this [./test_aml_mock_fixtures_default.py](./test_aml_mock_fixtures_default.py) file under your tests directory
+
+    This code mocks frequently used objects in AML SDK and returns them as fixtures so that they can be reused in other code.
+
+1. Import predefined fixture to your test code
+
+    ```python
+    from ml_service.tests.pipelines.test_aml_mock_fixtures_default import aml_pipeline_mocks
+    ```
+
+    > Note: Depending on the location you are referencing the fixture file, this path 'ml_service.tests.pipelines.test_aml_mock_fixtures_default' may need to be different.
+
+1. Pass 'aml_pipeline_mocks' as a parameter to your unit test code and load mock objects from fixture by referencing the code below.
+
+    ```python
+    def test_build_data_processing_os_cmd_pipeline_happy_path(mocker: MockFixture, aml_pipeline_mocks):  
+    
+    # Load mocks from fixture
+    (workspace, aml_compute, mock_workspace_get, mock_pipeline_publish) =\
+        aml_pipeline_mocks
+    ```
+
+    Each mock object can be accessed by listing them again cause it uses tuple notation.
+
+1. Assertion using mocks
+
+    ```python
+    # Check if the correct workspace retrieved
+    mock_workspace_get.assert_called_with(name=workspace_name,
+                                          resource_group=resource_group,
+                                          subscription_id=subscription_id)
+
+    # Check if Pipeline publish was called with arguments
+    mock_pipeline_publish.assert_called_once_with(name=preprocessing_pipeline_name,
+                                                  description="Data preprocessing"
+                                                              " OS cmd pipeline",
+                                                  version=build_id)
+    ```
+
+    This code shows how to make an assertion using an imported mock objects.
+
+    The full unit test code can be found at [test_build_data_processing_os_cmd_pipeline.py](/samples/non-python-preprocess/ml_service/tests/pipelines/test_build_data_processing_os_cmd_pipeline.py).
+
+### Fixture for mocking AML SDK with Env util
+
+> [This fixture](./test_aml_mock_fixtures_env.py) is **dependent on the ml_service.util.env_variables.Env class**
+used in [non-python-preprocess sample](/samples/non-python-preprocess/ml_service/util/env_variables.py).
+
+This fixture is for the case of referencing environment variables using env util.
+
+### How to use
+
+TODO
```

Comment:
You can basically copy the advise from the independent fixture and change the comment that workspace_name, rg, sub_id are now changed at a different place (within mock for EnvUtil)